tests/model/test_model.py

```diff
@@ -1528,6 +1528,57 @@ def test_fetch_raw_message_content(
         assert self.display_error_if_present.called == display_error_called
         assert return_value == expected_raw_content
 
+    @pytest.mark.parametrize(
+        "response, expected_read_receipts, feature_level",
+        [
+            case(
+                {"msg": "", "result": "success", "user_ids": [3, 7, 9]},
+                [3, 7, 9],
+                139,
+                id="valid_ZFL_with_valid_response",
+            ),
+            case(
+                {"msg": "Invalid message(s)", "result": "error"},
+                None,
+                137,
+                id="valid_ZFL_with_invalid_response",
+            ),
+        ],
+    )
+    def test_fetch_message_read_receipt_user_ids_valid_zfl(
+        self,
+        mocker,
+        model,
+        expected_read_receipts,
+        response,
+        feature_level,
+        message_id=1,
+    ):
+        model.server_feature_level = feature_level
+        self.client.call_endpoint.return_value = response
+        result = model.fetch_message_read_receipt_user_ids(message_id)
+        assert result == expected_read_receipts
+
+    @pytest.mark.parametrize(
+        "feature_level",
+        [
+            case(
+                136,
+                id="invalid_ZFL_with_response",
+            ),
+        ],
+    )
+    def test_fetch_message_read_receipt_user_ids_invalid_zfl(
```

Comment:
Nit: When naming test functions, we've aimed to add a second `_` to separate the function name from the situation we're handling.

I need to document this :)

Reply:
Changing it to `test_fetch_message_read_receipt_user_ids__invalid_zfl`

Should I also change the valid case function also to `test_fetch_message_read_receipt_user_ids__valid_zfl`?

Reply:
I just noticed, not many test functions with invalid/error cases have double underscores. I take it its a new thing?

Reply:
I've not done a survey on how much of the existing code has this. We don't have a testing 'style guide' right now - and many early tests were intended to get any testing setup! We've refactored and adjusted things quite a lot :)

This is more of a final point to note, ie. a very small thing - it's more important we get the tests there and functional - but things like making them more readable will help others.